game/src/character/objects/PasswordChest.java

```diff
@@ -0,0 +1,87 @@
+package character.objects;
+
+import basiselements.hud.TextButtonListener;
+import collision.CharacterDirection;
+import collision.Collidable;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import controller.ScreenController;
+import tools.Point;
+
+public class PasswordChest extends TreasureChest {
+
+    private String password;
+    private PasswordInputUI ui;
+    private int attemptCounter = 0;
+    private boolean correctPassword = false;
+    private boolean interacting = false;
+    private TextButtonListener okListener =
+            new TextButtonListener() {
+                @Override
+                public void clicked(InputEvent event, float x, float y) {
+                    onOK();
+                }
+            };
+
+    private TextButtonListener exitListener =
+            new TextButtonListener() {
+                @Override
+                public void clicked(InputEvent event, float x, float y) {
+                    onExit();
+                }
+            };
+
+    public PasswordChest(Point position, String password) {
+        super(position);
+        this.password = password;
+    }
+
+    private void onOK() {
+        if (password.equals(ui.passwordInput.getText())) {
+            correctPassword = true;
+            if (!isOpen) {
```

Comment:
Idee: damit der Code nicht doppelt existiert:
In `TreasureChest`eine protect void open Methode hinzufügen und dort
```
currentAnimation = opening;
inventory.forEach(i -> i.collect());
inventory.clear();
isOpen = true;
```

und dann hier (und in der Kollision Methode von TreasureChest) 
```
 if (!isOpen) open();`